### PR TITLE
Reorganize and refactor flask application

### PR DESCRIPTION
**Pull request by nc-sauleanf**
### Related PRs
Are there any Related PRs?
- [ ] Yes
- [x] No

### Feature ID
Please see this [confluence page](https://confluence.internal.nextcapital.com/display/~sauleanf/NC+Deployment+Log+Application+Current+and+Upcoming+Features) for information about feature 88182690.

### Testing
Did you write tests?
- [x] Yes
- [ ] No

### CODE REVIEW

File: config/database.example.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+development:
+  mongo:
+    uri: "mongodb://localhost:27017"
+    database: "development_deployment_log"
+
+test:
+  mongo:
+    uri: "mongodb://localhost:27017"
+    database: "test_deployment_log"
```
Comment (nc-yanga) [inline config/database.example.yml:9]: stop at the end -- might need to add newline?

File: blueprints/users.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from flask import Blueprint, jsonify, request
```
Comment (nc-yanga) [inline blueprints/users.py:1]: good idea to import blueprints!

File: make_graphs.py [hunk 3]
```diff
@@ -4,30 +4,41 @@
...
-    if (
-        cur_day != prev_day
-    ):  # if we have reached a new day, add a new entry to each list
+    # if we have reached a new day, add a new entry to each list
+    if cur_day != prev_day:
         for env in env_names:
             environment2num_deployments[env].append(
                 environment2num_deployments[env][-1]
             )
         prev_day = cur_day
```
Comment (nc-yanga) [inline make_graphs.py:26]: Would there ever be a situation where the days are not in the right order?

File: app_config.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import os
+import yaml
+
+environments = ["development", "test", "development"]
+config = {environment: {} for environment in environments}
+
+for file in os.listdir("config"):
+    with open(os.path.join("config", file)) as config_file:
```
Comment (nc-yanga) [inline app_config.py:8]: which config file?

File: blueprints/deployments.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
+from flask import Blueprint, jsonify, request, Response, make_response, send_file
```
[the remark this answers is not in the document]
Comment (nc-sauleanf) [reply blueprints/deployments.py:1]: It is pretty much just the old app.py minus the routes

File: blueprints/users.py (hunk 2 above)
Comment (nc-sauleanf) [reply blueprints/users.py:1]: 👍

File: make_graphs.py (hunk 3 above)
Comment (nc-sauleanf) [reply make_graphs.py:26]: That is the old graphing script. I don't know if we will reuse that script.

File: tests/blueprints/test_deployments.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        self.assertGreaterEqual(
+            body.items(),
+            {
+                "page": 1,
+                "per_page": self.per_page,
+                "number_of_pages": self.number_of_pages
+             }.items()
+        )
+
+    def test_create(self):
```
Comment (nc-yanga) [inline tests/blueprints/test_deployments.py:57]: We should prob create a larger variety of tests at some point, but looks good

**Pull request merged**